packages/kbn-apm-config-loader/src/types.ts

```diff
@@ -10,4 +10,20 @@
 // but it's not exported, and using ts tricks to retrieve the type via Parameters<ApmAgent['start']>[0]
 // causes errors in the generated .d.ts file because of esModuleInterop and the fact that the apm module
 // is just exporting an instance of the `ApmAgent` type.
-export type ApmAgentConfig = Record<string, any>;
+export interface ApmAgentConfig {
```

Comment:
RUM exports the `AgentConfigOptions` - https://github.com/elastic/apm-agent-rum-js/blob/cb0fada417d815eccac018e3d9c495bb3bcadbe2/packages/rum/src/index.d.ts#L27

We can use that here and add extra fields for Node.js ones.

Reply:
I'm a little concerned about rum-agent specific config ending up in that type and being a bit misleading. I think I'd much prefer to consume the type from the package we're actually going to be using the object from, and short of that a custom type which only implements the properties we know are supported feels more likely to work in the long term... Thoughts?

Reply:
Yeah makes sense, I am fine with waiting till Node.js agent officially exports typings for the config.